### DISCUSSION TITLE
(eks) addManifest with multiple documents/resources fails in the lambda

### DISCUSSION THREAD

**Discussion by chalford** [aws/aws-cdk#22542] [Q&A]
Not sure if this is a bug or user error, but I'm trying to use `addManifest` on an EKS cluster to deploy a manifest YAML file that contains multiple resources. Reading the docs https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-eks.KubernetesManifest.html suggests that it supports "any number of child resources".

I've tried two approaches here. First, was to define, in TypeScript, two objects in an array that is passed to addManifest, like this:
```
cluster.addManifest(
  'eks-example-manifest',
  [{
    apiVersion: 1,
    kind: Namespace,
    ...
  },
  {
    apiVersion: 1,
    kind: Deployment,
    ...
  }]);
```
Looking in the logs for the lambda, I get an error with validating the manifest. I can see in the log that that manifest passed to the lambda now has an extra pair of square brackets around it - for some reason it's been inserted the the only element in another array.

The second approach is to load the same manifest example as YAML:
```
const exampleManifest = yaml.loadAll(
  fs.readFileSync('lib/eks-console-full-access.yaml', 'utf8')
);
cluster.addManifest(
  'eks-example-manifest',
  exampleManifest);
```
This causes exactly the same problems as the first approach.

Can anyone see if I'm missing something? Thank you in advance for your help.

**Comment by pahud** [comment]
Sounds like a bug. Can you provide your yaml file so we can try reproduce and validate it? Also, can you share the full error messages or screenshot of Lambda logs?

**Comment by julian-saunders** [comment]
I have noticed something similar.  I'm using the manifest found here:

https://raw.githubusercontent.com/aws-samples/amazon-cloudwatch-container-insights/latest/k8s-deployment-manifest-templates/deployment-mode/daemonset/container-insights-monitoring/fluent-bit/fluent-bit.yaml

The following fails with a template validation error:
```
const fluentBitYaml = yaml.loadAll(fs.readFileSync('monitoring/fluent-bit.yaml', 'utf8'));
cluster.addManifest('fluentBit', fluentBitYaml);
```

If however if I reduce the manifest to a single resource and use yaml.load not yaml.loadAll then the manifest is deployed successfully.

**Comment by julian-saunders** [comment]
I found adding three fullstops/dots/periods (...) in front of the manifest parsed to addManifest overcomes the issue:

`const fluentBit = cluster.addManifest('fluentBit', ...fluentBitYaml);`

**Reply by 44int** [reply]
This workaround worked for me too.

**Reply by pranaymodukuru** [reply]
Adding three periods throws this error for me 
> error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'Record<string, any>'.

**Reply by pranaymodukuru** [reply]
for three dots to work, the output of loadAll has to be typecasted as `[Record<string, any>]`
```
const fluentBitYaml = yaml.loadAll(fs.readFileSync('monitoring/fluent-bit.yaml', 'utf8')) as [Record<string, any>];
cluster.addManifest('fluentBit', ...fluentBitYaml);
```
